portal-impl/src/com/liferay/portal/template/TemplatePortletPreferences.java

```diff
@@ -34,29 +33,38 @@
 	public String getPreferences(Map<String, Object> preferences)
 		throws ReadOnlyException {
 
-		PortletPreferencesImpl portletPreferencesImpl =
-			new PortletPreferencesImpl();
+		StringBundler sb = new StringBundler();
+
+		sb.append("<portlet-preferences>");
 
 		for (Map.Entry<String, Object> entry : preferences.entrySet()) {
-			Object value = entry.getValue();
+			sb.append("<preference><name>");
+			sb.append(entry.getKey());
+			sb.append("</name>");
+
+			String[] values = _NULL_VALUE;
```

Comment:
The old TemplatePortletPreferences logic only possibly takes in String and String[] values, now with the object value map and the instanceOf checking, we could possibly take in other types, which is changing the expected logic, we should at least prevent that. I will make some changes for that.